### ISSUE TITLE
// .*DO NOT EDIT.?$

### ISSUE THREAD

**Issue by butuzov** [microsoft/vscode-go#3243]
Please answer these questions before submitting your issue. Thanks!

### What version of Go, VS Code & VS Code Go extension are you using?
- Run `go version` to get version of Go
	- `go version go1.14.2 darwin/amd64`
- Run `code -v` or `code-insiders -v` to get version of VS Code or VS Code Insiders
	- `1.45.1`
	- `5763d909d5f12fe19f215cbfdd29a91c0fa9208a`
	- `x64`

### Describe the bug
`vscode-go` only notice generated files with `// Code generated ... DO NOT EDIT` pattern at the first line.  There are actually two problems with this.
1. ~ `// Code generated ... DO NOT EDIT` described by `go help generate`  and in [https://golang.org/s/generatedcode](https://golang.org/s/generatedcode) is more *desirable* state than, actual fact. See Examples (`1`, `2`)
2. Quoting Rob Pike `The text may appear anywhere in the file.`, not just on the first line.

#### Examples
1.  some examples from `golang/go` 
```
find . -name "*.go" -exec cat "{}" \;  | grep -E "^// .*?DO NOT EDIT" | uniq -c | sort -nr 
...
155 // THIS FILE IS GENERATED BY THE COMMAND AT THE TOP; DO NOT EDIT
 18 // Created by cgo -godefs - DO NOT EDIT
  5 // DO NOT EDIT
  1 // GENERATED BY make_perl_groups.pl; DO NOT EDIT.
```
2. Search for `go` files on a GitHub gives more examples of the same behavior

### Propose 
What community really follows is a `DO NOT EDIT` phrase in a comment. Can we also have this a trigger to show `This file seems to be generated. DO NOT EDIT.` error message?

**Comment by hyangah** [comment]
Wish https://github.com/golang/go/issues/13560 made any progress in this problem domain.

@stamblerre what does `gopls` do to detect generated code?

**Comment by butuzov** [comment]
@hyangah  does `vscode-go` going to require `gopls`  in the next releases?

**Comment by hyangah** [comment]
@butuzov `gopls` is still in beta. So we will keep the option to use this extension without `gopls` open for a while.  But we hope to consider `gopls` and `dlv dap` first if the features can be provided through LSP and DAP.

**Comment by butuzov** [comment]
@hyangah got it. Anyway, my PR with fixes ready for review.

**Comment by stamblerre** [comment]
@hyangah: Here's the code `gopls` uses for detecting generated files: https://github.com/golang/tools/blob/master/internal/lsp/source/util.go#L144.

@butuzov: Thanks for putting together a PR. We are actually currently migrating to a new repository, so we are only accepting PRs in https://github.com/golang/vscode-go right now. We'll have to kindly request that you send your PR there instead. You can see more details here: https://github.com/microsoft/vscode-go#we-are-moving. I'm sorry about the inconvenience.

**Comment by butuzov** [comment]
closed in favour of the golang/vscode-go#68
